From a3c340db5c1c4707d5e6344564f322ed831320cd Mon Sep 17 00:00:00 2001
From: Robert Mustacchi <rm@joyent.com>
Date: Mon, 3 Apr 2017 17:53:37 +0000
Subject: [PATCH] OS-5970 Want NVMe 1.2 support Reviewed by: Hans Rosenfeld
 <hans.rosenfeld@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/man/man7d/nvme.7d         | 4 ++--
 usr/src/uts/common/io/nvme/nvme.c | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/usr/src/man/man7d/nvme.7d b/usr/src/man/man7d/nvme.7d
index 03420fd19d..8043d53b2a 100644
--- a/usr/src/man/man7d/nvme.7d
+++ b/usr/src/man/man7d/nvme.7d
@@ -11,7 +11,7 @@
 .\"
 .\" Copyright 2016 Nexenta Systems, Inc.  All rights reserved.
 .\"
-.Dd May 13, 2016
+.Dd Mar 30, 2017
 .Dt NVME 7D
 .Os
 .Sh NAME
@@ -48,7 +48,7 @@ to attach to devices supporting newer version of the NVMe
 specification.
 The default value is 1, limiting
 .Nm
-to work with devices up to specification version 1.1.
+to work with devices up to specification version 1.2.
 .It Va ignore-unknown-vendor-status
 This can be set to 1 to allow
 .Nm
diff --git a/usr/src/uts/common/io/nvme/nvme.c b/usr/src/uts/common/io/nvme/nvme.c
index 8258d7964b..9809410a09 100644
--- a/usr/src/uts/common/io/nvme/nvme.c
+++ b/usr/src/uts/common/io/nvme/nvme.c
@@ -18,9 +18,9 @@
 /*
  * blkdev driver for NVMe compliant storage devices
  *
- * This driver was written to conform to version 1.1b of the NVMe specification.
- * It may work with newer versions, but that is completely untested and disabled
- * by default.
+ * This driver was written to conform to version 1.2.1 of the NVMe
+ * specification.  It may work with newer versions, but that is completely
+ * untested and disabled by default.
  *
  * The driver has only been tested on x86 systems and will not work on big-
  * endian systems without changes to the code accessing registers and data
@@ -207,7 +207,7 @@
 
 /* NVMe spec version supported */
 static const int nvme_version_major = 1;
-static const int nvme_version_minor = 1;
+static const int nvme_version_minor = 2;
 
 /* tunable for admin command timeout in seconds, default is 1s */
 static volatile int nvme_admin_cmd_timeout = 1;
-- 
2.21.0

